lms/static/js/learner_dashboard/views/entitlement_unenroll_view.js

```diff
@@ -0,0 +1,158 @@
+                    });
+                },
+
+                handleTrigger: function(event) {
+                    var $trigger = $(event.target),
+                        courseName = $trigger.data('courseName'),
+                        courseNumber = $trigger.data('courseNumber'),
+                        isRefundable = $trigger.data('entitlementIsRefundable'),
+                        apiEndpoint = $trigger.data('entitlementApiEndpoint');
+
+                    this.resetModal();
+                    this.setTrackInfo(courseName, courseNumber);
+                    this.setRefundInfo(isRefundable);
+                    this.setSubmitData(apiEndpoint, isRefundable);
+
+                    if (window.edx && window.edx.dashboard && window.edx.dashboard.dropdown) {
+                        window.edx.dashboard.dropdown.toggleCourseActionsDropdownMenu(event);
+                        this.$el.css('position', 'fixed');
+                    }
+                },
+
+                handleSubmit: function() {
+                    var apiEndpoint = this.$submitButton.data('entitlementApiEndpoint'),
+                        isRefundable = this.$submitButton.data('entitlementIsRefundable');
+
+                    if (apiEndpoint === undefined || isRefundable === undefined) {
+                        this.setError(gettext('Error: cannot process unenrollment request.'));
+                        return;
+                    }
+
+                    $.ajax({
+                        url: apiEndpoint,
+                        method: 'DELETE',
+                        complete: this.onComplete.bind(this)
+                    });
+                },
+
+                resetModal: function() {
+                    this.$submitButton.removeData();
+                    this.$submitButton.prop('disabled', false);
+                    this.$trackInfo.empty();
+                    this.$refundInfo.empty();
+                    this.$errorInfo.css('display', 'none');
+                    this.$errorInfo.empty();
+                },
+
+                setError: function(message) {
+                    this.$submitButton.prop('disabled', true);
+                    this.$errorInfo.empty();
+                    HtmlUtils.setHtml(
+                        this.$errorInfo,
+                        message
+                    );
+                    this.$errorInfo.css('display', 'block');
+                },
+
+                setTrackInfo: function(courseName, courseNumber) {
+                    this.$trackInfo.empty();
+                    HtmlUtils.setHtml(
+                        this.$trackInfo,
+                        HtmlUtils.interpolateHtml(
+                            gettext('Are you sure you want to unenroll from {courseName} ({courseNumber})?'),
+                            {
+                                courseName: HtmlUtils.joinHtml(
+                                    HtmlUtils.HTML('<span id="entitlement-unenrollment-course-name">'),
```

Comment:
Open question: why is this span necessary (it was copied from the original modal logic)?